File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    list = [chr(i) for i in range(97, 123)]
    for i in skip:
        list.remove(i)
    
    def find(list, s, index):
        order = (list.index(s) + index) % len(list)
        return list[order]
    
    for i in s:
        answer += find(list, i, index)
    
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    const char = []
    const aCode = "a".charCodeAt();
    const zCode = "z".charCodeAt();
    for (let i = aCode; i <= zCode; i++) {
        const c = String.fromCharCode(i)
        if ([...skip].includes(c)) continue
        char.push(c)
    }
    
    const answer = [...s].map((el) => {
        const newIdx = (char.indexOf(el) + index) % char.length
        return char[newIdx]
    }).join("")
    return answer;
}